app/cdap/services/helpers.js

```diff
@@ -527,23 +527,29 @@ function categorizeGraphQlErrors(error) {
     errorsByOrigin[GENERIC_ERROR_ORIGIN] = error.message;
   }
 
-  graphQLErrors.forEach(error => {
-    const errorOrigin = objectQuery(error, 'extensions', 'exception', 'errorOrigin') || GENERIC_ERROR_ORIGIN;
-    if (errorsByOrigin.hasOwnProperty(errorOrigin)) {
-      errorsByOrigin[errorOrigin].push(error.message);
-    }
-    else {
-      errorsByOrigin[errorOrigin] = [error.message];
-    }
-  });
+  if (Array.isArray(graphQLErrors)) {
```

Comment:
Since this is a fix for an error we're not sure how to reproduce, I'd suggest adding a comment saying so here. `// TODO: Array.isArray() is guarding against a bug we haven't reproed. error is the {error} value from apollo.useQuery(). When is this false?` - I also might add an `else { console.error(...); }`. I don't think it's the worst thing to commit a partial fix like this, but I think it's important that the code not be silent about the reasoning behind it.